tests/test_database.py

```diff
@@ -238,6 +238,32 @@ async def test_branch_drop_recreate(self):
         finally:
             await tb.drop_db(self.con, 'test_db_drop')
 
+    async def test_branch_drop_disconnect(self):
+        if not self.has_create_database:
+            self.skipTest("create branch is not supported by the backend")
+
+        await self.con.execute('CREATE EMPTY BRANCH test_db_disconnect;')
+        conn = await self.connect(database='test_db_disconnect')
+
+        try:
+            dbname = await conn.query(
+                'SELECT sys::get_current_database();')
+            self.assertEqual(dbname, ['test_db_disconnect'])
+
+            # Drop branch while the frontend connection is active
+            await self.con.execute(
+                'DROP BRANCH RESET CONNECTIONS test_db_disconnect'
+            )
+
+            # The frontend connection should be closed by the server now
+            self.assertTrue(conn.is_closed())
```

Comment:
Are you confident that this will be reflected in the client state already?

Reply:
Yeah, the test passed at [b851e5c](https://github.com/edgedb/edgedb/actions/runs/7791023176/job/21246262435#step:18:222) - could be flaky but I haven't met